lib/Performance/Probability.pm

```diff
@@ -0,0 +1,306 @@
+package Performance::Probability;
+
+use 5.010;
+use strict;
+use warnings;
+use Moo;
+
+use Date::Utility;
+use Math::BivariateCDF;
+use Math::Gauss::XS;
+
+our $VERSION = '0.01_1';
+
+=head1 NAME
+
+Performance::Probability - The performance probability is a likelihood measure of a client reaching his/her current P&L. 
+
+=head1 VERSION
+
+0.01_1
+
+=head1 SYNOPSYS
+
+=head1 DESCRIPTION
+
+=cut
+
+=head1 ATTRIBUTES
+
+=head2 payout
+
+Payout 
+
+=cut
+
+has payout => (
+    is       => 'ro',
+    required => 1,
+);
+
+=head2 bought_price
+
+Bought price
+
+=cut
+
+has bought_price => (
+    is       => 'ro',
+    required => 1,
+);
+
+=head2 pnl
+
+PnL
+
+=cut
+
+has pnl => (
+    is       => 'ro',
+    required => 1,
+);
+
+=head2 type
+
+Type of contract. Call or Put.
```

Comment:
Actually this is **types** (plural). And from the single usage

```
if ($self->type->[$i] ne $self->type->[$j])
```

they might be everything, not only `call` or `put`, but also `binary.com` and `google.com`